providers/implementations/storemgmt/file_store.c

```diff
@@ -485,6 +485,8 @@ static int file_set_ctx_params(void *loaderctx, const OSSL_PARAM params[])
         BIO_snprintf(ctx->_.dir.search_name, sizeof(ctx->_.dir.search_name),
                      "%08lx", hash);
         X509_NAME_free(x509_name);
+        if (hash == 0)
```

Comment:
This is wrong as there is a small 1 in 2^64 chance that the real name hash value will be actually 0.

Reply:
Perhaps add ERR_peek_error() != 0 condition?

Reply:
> This is wrong as there is a small 1 in 2^64 chance that the real name hash value will be actually 0.

I had also thought about the risk of the hash being legitimately 0.
When looking at the implementation of `X509_NAME_hash()`, one can see that the chance is actually 1 in 2^32,
and the value 0 is actually returned on error as well, which is a bad design.

> Perhaps add ERR_peek_error() != 0 condition?

This would work only if the error queue was/made empty before, which is not guaranteed/conservative,
and in either case not elegant.
Since I believe that "(unsigned) long" is meanwhile guaranteed to be actually 64 bits wide,
I've resorted to letting this function (and consequently also `X509_NAME_hash_old()`) return `0x100000000L` in case of error.

Reply:
I don't think you can safely assume that long is 64 bits.
See, for example, the commit message for 66ecfb5437b22664a4358e8de6da942727f5fb11

Reply:
I see, but then `0x100000000` collapses to `0`, which AFAICS is still the best one can do 
without changing the type of the function (in some not backward compatible way).